lib/GeoExt/data/Print3Provider.js

```diff
@@ -0,0 +1,374 @@
+    },
+
+    /** private: method[validatePrintOptions]
+     */
+    validatePrintOptions: function(map, pages, options) {
+        if (pages.length != 1) {
+            this.fireEvent("printexception", this, response);
+            return false;
+        }
+        return true;
+    },
+
+    /** api: method[requestPrint]
+     */
+    requestPrint: function(map, pages, encodedLayers, encodedOverviewLayers, encodedLegends, callback) {
+        var outputFormat = this.outputFormat ? this.outputFormat.get("name") : this.defaultOutputFormatName;
+        var page = pages[0];
+
+        var jsonData = {
+            layout: this.layout.get("name"),
+            outputFormat: outputFormat,
+            attributes: this.customParams
+        };
+
+        var mapAttributes = this.getAttributesByType("MapAttributeValues");
+        var overviewMapAttributes = this.getAttributesByType("OverviewMapAttributeValues");
+        var legendAttributes = this.getAttributesByType("LegendAttributeValue");
+
+        jsonData.attributes[mapAttributes[0].name] = {
+            projection: map.getProjection(),
+            dpi: this.dpi.get("value"),
+            rotation: page.rotation,
+            center: [page.center.lon, page.center.lat],
+            scale: page.scale.get("value"),
+            longitudeFirst: true,
+            layers: encodedLayers
+        };
+
+        if (overviewMapAttributes && encodedOverviewLayers) {
+            jsonData.attributes[overviewMapAttributes[0].name] = {
+                layers: encodedOverviewLayers
+            };
+        }
+
+        if (legendAttributes) {
+            jsonData.attributes[legendAttributes[0].name] = {
+                "classes": encodedLegends
+            };
+        }
+
+        Ext.Ajax.request({
+            url: this.url + "/report." + outputFormat,
+            method: 'POST',
+            jsonData: jsonData,
+            headers: { "Content-Type": "application/json; charset=" + this.encoding },
+            success: function(response) {
+                var ref = Ext.decode(response.responseText).ref;
+                callback(ref);
+            },
+            failure: function(response) {
+                this.fireEvent("printexception", this, response);
+            },
+            params: this.baseParams,
+            scope: this
+        });
+    },
+
+    getStatus: function(ref, callback) {
+        Ext.Ajax.request({
+            url: this.url + '/status/' + ref + '.json',
+            method: 'GET',
+            success: function(response) {
+                var current_status = Ext.decode(response.responseText);
+                callback(ref, true, current_status);
+            },
+            failure: function(response) {
+                callback(ref, false);
+                this.fireEvent("printexception", this, response);
+            },
+            params: this.baseParams,
+            scope: this
+        });
+    },
+
```

Comment:
`getStatus` and `getDownloadURL` Are useful method for people that used the MapFishPrintv3 it's why it's not used internally, and it's important to document them :-)